bioagents/qca/qca_module.py

```diff
@@ -0,0 +1,289 @@
+            sources = [str(k.data) for k in source_arg.data]
+
+        if reltype_arg is None or len(reltype_arg.data) < 1:
+            relation_types = None
+        else:
+            relation_types = [str(k.data) for k in reltype_arg.data]
+
+        #qca = QCA()
+        #source_names = ["IRS1"]
+        #target_names = ["SHC1"]
+
+        has_path = self.qca.has_path(sources, targets)
+
+        reply_content = KQMLList.from_string(
+            '(SUCCESS :haspath (' + str(has_path) + '))')
+
+        return reply_content
+
+    def respond_find_qca_path_local(self, content_list):
+        target_arg = content_list.get_keyword_arg('TARGET')
+        targets = []
+        source_arg = content_list.get_keyword_arg('SOURCE')
+        sources = []
+
+        if len(target_arg.data) < 1:
+            raise ValueError("Target list is empty")
+        else:
+            targets = [str(k.data) for k in target_arg.data]
+
+        if len(source_arg.data) < 1:
+            raise ValueError("Source list is empty")
+        else:
+            sources = [str(k.data) for k in source_arg.data]
+
+        directedPaths = DirectedPaths()
+        source = ['EGFR']
+        target = ['MAP2K1', 'MAP2K2']
+        max_number_of_paths = 5
+        return_this = []
+        with open('RAS_Machine.cx', 'rb') as network_file:
+            network = json.load(network_file)
+            found_directed_path = dict(data=directedPaths.findDirectedPaths(network, sources, targets, npaths=max_number_of_paths))
+            paths_english = found_directed_path.get("data").get("forward_english")
+
+            for path in paths_english:
+                path_statement = ""
+                for n in path:
+                    if type(n) is str:
+                        path_statement += n + " "
+                    elif type(n) is unicode:
+                        path_statement += str(n) + " "
+                    elif type(n) is dict:
+                        tmp = n.itervalues().next()
+                        if "interaction" in tmp:
+                            path_statement += tmp.get("interaction") + " "
+                        elif "INDRA statement" in tmp:
+                            path_statement += tmp.get("INDRA statement") + " "
+                        else:
+                            path_statement += tmp + " "
+                    else:
+                        path_statement += " error message: edge missing "
+                return_this.append(str(path_statement))
+
+            print return_this
+
+        print 'Here is the input %s' % target_arg
+
+        path_statements = "~ ".join(return_this)
+
+        reply_content = KQMLList.from_string(
+            '(SUCCESS :paths (' + path_statements + '))')
+
+        return reply_content
+
+        #reply_msg = KQMLPerformative('reply')
+        #reply_msg.set_parameter(':content', reply_content)
+        #self.reply(msg, reply_msg)
+
+    def respond_find_qca_path_template(self, content_list):
```

Comment:
This seems like something copied from the DTDA code, can it be removed or is it needed here?